### PR TITLE
ForwardDiff directly on all non-C solvers

### PR DESCRIPTION
**Pull request by lxvm**
This should fix one of the last problems in https://github.com/SciML/SciMLExpectations.jl/pull/148

In this pr we ForwardDiff directly on all non-C solvers since there are fewer restrictions on integrand types

## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by ArnoStrouwen** [comment]
I can't find the missing link that the Documenter CI is complaing about?

**Comment by ChrisRackauckas** [comment]
I just saw this in Symbolics too. @mortenpi is this a Documenter bug?

File: docs/src/basics/FAQ.md [hunk 1]
```diff
@@ -58,6 +59,13 @@ because that is what lower-level packages implement.
 Fixed quadrature rules from other packages can be used with `QuadratureRule`.
 Otherwise, feel free to open an issue or pull request.
 
+## My integrand works with algorithm X but fails on algorithm Y
+
+While bugs are not out of the question, certain algorithms, especially those implemented in C, are not compatible with arbitrary Julia types and have to return specific numeric types or arrays thereof.
+In some cases, such as [`ArblibJL`](@ref), it is also expected that the integrand work with a custom quadrature point type.
+Moreover, some algorithms, such as [`VEGAS`](@ref), only support scalar integrands.
+For more details see the [solver page](solvers).
```
Comment (mortenpi) [inline docs/src/basics/FAQ.md:67]: I presume you mean

```
┌ Error: invalid local link/image: file does not exist in src/basics/FAQ.md
│   link =
│    @ast MarkdownAST.Link("solvers", "") do
│      MarkdownAST.Text("solver page")
│    end
│    
└ @ Documenter ~/.julia/packages/Documenter/1HwWe/src/utilities/utilities.jl:44
```

In which case it's this one.

Comment (ArnoStrouwen) [reply docs/src/basics/FAQ.md:67]: It should likely just be
```suggestion
For more details see the [solver page](@id solvers).
```

Comment (mortenpi) [reply docs/src/basics/FAQ.md:67]: Should be `@ref`:

```suggestion
For more details see the [solver page](@ref solvers).
```

Comment (ArnoStrouwen) [reply docs/src/basics/FAQ.md:67]: Yeah... 12 hour time difference with NZ will do that.

**Pull request merged**